### PR TITLE
Restructure wallet-connect namespaces

### LINKED ISSUE #21166: Restructure wallet-connect namespaces
# Problem
Following the initial WalletConnect implementation and the rush to release, a restructuring became necessary to stabilize the future development of the feature.

## Implementation
We have a few namespaces that became too large, mixing many flows together (e.g. `events`, `subs`, `utils`, `core`, etc.). Ideally, we should split them based on the context (e.g. proposal, sessions, etc.).

## Acceptance Criteria

1. No regressions following the restructuring
2. No changes to existing functionality or refactors, just moving functionality into different namespaces (otherwise will be difficult to track changes and test.

### PR DESCRIPTION
**Pull request by clauxx**
fixes #21166

### Summary
Following the initial WalletConnect implementation and the rush to release, a restructuring became necessary to stabilize the future development of the feature.
### Review notes
There are no functional changes other than the namespace names (which cause formatting changes sometimes) and moving of functions into different namespaces.
### Testing notes
Need to make sure the main flows work as expected, as all the wallet-connect code was touched. There are no functional changes though, so things should work as they did before.

#### Platforms

- Android
- iOS

##### Functional

- wallet connect

status: ready

### CODE REVIEW

**Review by shivekkhurana — APPROVED** [review]

File: src/status_im/contexts/wallet/wallet_connect/utils/data_transformations.cljs [hunk 1]
```diff
@@ -0,0 +1,73 @@
+(ns status-im.contexts.wallet.wallet-connect.utils.data-transformations
```
Comment (mohsen-ghafouri) [inline src/status_im/contexts/wallet/wallet_connect/utils/data_transformations.cljs]: in other places we named this kind of file `data-store`, maybe we can use the same name here?

**Review by mohsen-ghafouri — APPROVED** [review]
Overall looks good to me. Thank you for the amazing work 🚀

File: src/status_im/contexts/wallet/wallet_connect/utils/data_transformations.cljs (hunk 1 above)
Comment (clauxx) [reply src/status_im/contexts/wallet/wallet_connect/utils/data_transformations.cljs]: thanks! i knew there was a "convention" in our project for these kind of files but couldn't find and example. Will rename it, cause it's also shorter.

Comment (clauxx) [reply src/status_im/contexts/wallet/wallet_connect/utils/data_transformations.cljs]: done :+1:

**Review by alwx — APPROVED** [review]

**Comment by clauxx** [comment]
@status-im/mobile-qa ready for testing

**Comment by pavloburykh** [comment]
@clauxx thanks for the PR. Please, take a look at the issue

### ISSUE 1 No account is selected by default in connection modal

**Steps:**

1. Scan wallet connect QR via universal scanner
2. See if account is selected within connection modal

**Actual result:** no account selected. 

photo_2024-09-06 13 43 45

**Expected result:** default account should be selected

**Comment by mohsen-ghafouri** [comment]
I will take a look to this issue

**Comment by mohsen-ghafouri** [comment]
Hey @mariia-skrypnyk could you please test again. thanks

**Comment by mariia-skrypnyk** [comment]
Hi @mohsen-ghafouri !
Thanks for your fix. Accounts could be changed successfully!

I see one more issue relates to account selection:

**ISSUE 2: user is navigated to Wallet Account without a connection**

**Steps:**

1. Login to app
2. Choose Account 1
3. Tap on WC icon
4. Scan QR code from test daap https://react-app.walletconnect.com/
5. Choose Account 2 within connection modal
6. Tap connect

**Actual result:**

User is redirected to Wallet account connection screen on Account 1 without a connection:

https://github.com/user-attachments/assets/6dbd907b-b35c-4320-b88e-30695403e7bb

**Expected result:** 

User should be redirected to Wallet account connection of chosen Account 2.

Design with navigation example is here https://www.figma.com/design/mLyiYPXqu8KetdQONOlZpV/dApp-Interactions?node-id=47-29943&node-type=frame&t=YPrT5xiOglqRZdyE-0

**Comment by mohsen-ghafouri** [comment]
Thank you @mariia-skrypnyk for checking this, could you please check if it's related to this PR or a general issue on dev, if so we can handle it in a follow up issue, as current PR is big enough to merge. 

also i'm not sure when user exactly is in this account we should let they change the selection or not.

**Comment by mariia-skrypnyk** [comment]
> Thank you @mariia-skrypnyk for checking this, could you please check if it's related to this PR or a general issue on dev, if so we can handle it in a follow up issue, as current PR is big enough to merge.
> 
> also i'm not sure when user exactly is in this account we should let they change the selection or not.

Thanks @mohsen-ghafouri !

This behaviour also exist on nightly.
I rerun tests and let you know.

**Comment by churik** [comment]
Failed tests are not related to PR, ready to merge

thanks @mariia-skrypnyk and @mohsen-ghafouri

**Comment by mohsen-ghafouri** [comment]
Hi @mariia-skrypnyk, I created this issue https://github.com/status-im/status-mobile/issues/21243 for [ISSUE 2](https://github.com/status-im/status-mobile/pull/21167#issuecomment-2340217297)

**Pull request merged**